ext/fetch/lib.rs

```diff
@@ -944,43 +985,62 @@ pub fn create_http_client(
     alpn_protocols.push("http/1.1".into());
   }
   tls_config.alpn_protocols = alpn_protocols;
+  let tls_config = Arc::from(tls_config);
+
+  let mut http_connector = HttpConnector::new();
+  http_connector.enforce_http(false);
+  let connector = HttpsConnector::from((http_connector, tls_config.clone()));
 
-  let mut headers = HeaderMap::new();
-  headers.insert(USER_AGENT, user_agent.parse().unwrap());
-  let mut builder = Client::builder()
-    .redirect(Policy::none())
-    .default_headers(headers)
-    .use_preconfigured_tls(tls_config);
+  let user_agent = user_agent
+    .parse::<HeaderValue>()
+    .map_err(|_| type_error("illegal characters in User-Agent"))?;
 
+  let mut builder =
+    hyper_util::client::legacy::Builder::new(TokioExecutor::new());
```

Comment:
Side question: does `hyper_util::client::legacy` imply that this code might get removed?

Reply:
Eventually, that is the goal. I'd like to redesign the `Client` to be more layers of middleware. There's a small sketch in the hyper-util repo. But not functioning code yet.

Since I knew I eventually wanted to redesign it, I named the "old" hyper 0.14 client as `legacy`. (That's also why it went to hyper-util, I knew it wasn't stable enough for `hyper` 1.0.)